grapher/core/EntityUrlBuilder.ts

```diff
@@ -1,19 +1,105 @@
+ *
+ * In short:
+ *
+ * Facebook replaces `%20` in URLs with `+`. Before this migration we encoded
+ * ["North America", "South America"] → "North%20America+South%20America".
+ * Facebook would turn this into "North+America+South+America" making the delimiters and spaces
+ * ambiguous.
+ *
+ * We chose ~ (tilde) because no entities existed in the database that contain that symbol, so the
+ * existence of that symbol could be used to detect legacy URLs.
+ *
+ */
 
 // Todo: ensure EntityName never contains the v2Delimiter
 
 const V1_DELIMITER = "+"
 export const ENTITY_V2_DELIMITER = "~"
 
+const isV1Param = (encodedQueryParam: string) => {
+    // No legacy entities have a v2Delimiter in their name,
+    // so if a v2Delimiter is present we know it's a v2 link.
+    return !decodeURIComponent(encodedQueryParam).includes(ENTITY_V2_DELIMITER)
+}
+
+const entityNamesFromV1EncodedParam = (
+    encodedQueryParam: string
+): EntityName[] => {
+    // need to use still-encoded URL params because we need to
+    // distinguish between `+` and `%20` in legacy URLs
+    return encodedQueryParam.split(V1_DELIMITER).map(decodeURIComponent)
+}
+
+const entityNamesToV2Param = (entityNames: EntityName[]): string => {
+    // Always include a v2Delimiter in a v2 link. When decoding we will drop any empty strings.
+    if (entityNames.length === 1) return ENTITY_V2_DELIMITER + entityNames[0]
+    return entityNames.join(ENTITY_V2_DELIMITER)
+}
+
+const entityNamesFromV2Param = (queryParam: string): EntityName[] => {
+    // Facebook turns %20 into +. v2 links will never contain a +, so we can safely replace all of them with %20.
+    return queryParam.split(ENTITY_V2_DELIMITER).filter((item) => item)
+}
+
+const migrateV1Delimited: UrlMigration = (url) => {
+    const { country } = url.encodedQueryParams
+    if (country !== undefined && isV1Param(country)) {
+        return url.updateQueryParams({
+            country: entityNamesToV2Param(
+                entityNamesFromV1EncodedParam(country)
+            ),
+        })
+    }
+    return url
+}
+
+/*
+ * Migration #2: Drop dimension keys from selected entities.
+ *
+ * Implemented: March 2021
+ *
+ * When plotting multiple variables on a chart, it used to be possible to pick which
+ * variable-entity pairs get plotted.
+ *
+ * For example, if you had a line chart with 3 variables: Energy consumption from Coal, Oil and Gas,
+ * then you (as a user) could select individual variable-entity pairs to plot:
+ *
+ * - France - Coal ("FRA-0")
+ * - France - Oil ("FRA-1")
+ * - France - Gas ("FRA-2")
+ * - ...
+ *
+ * The index of the dimension was appended to the entity (e.g. 0 for Coal).
+ *
+ * We dropped this feature in March 2021 in order to simplify the selection-handling logic, and it
+ * was also, in most cases, not desirable to present users with variable-entity options.
+ *
+ * We will likely re-introduce this in the future, architected better and with the option of which
```

Comment:
nit: I would like to know what the rest of the comment says :(

Reply:
haha well if you want to find out you can just